### PR TITLE
hack: enforce conftest and yamllint failures

### PR DESCRIPTION
**Pull request by spiffxp**
Add more enforcement to hack/verify.sh via:
- failing on `yamllint` rules that were previously warnings (`new-line-at-end-of-file` and `trailing-spaces`) to bring us to parity with the `yamllint` enforcment used for kubernetes/test-infra
- failing on `conftest` failures based on Open Policy Agent policies defined in policies/ 

Also tuned down the output noise from `verify-shellcheck.sh`

Part of https://github.com/kubernetes/k8s.io/issues/1734

### CODE REVIEW

File: hack/verify-conftest.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+REPO_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd )"
+
+function usage() {
+    echo >&2 "Usage: $0"
+    exit 1
+}
+
+function ensure_dependencies() {
```
Comment (aojea) [inline hack/verify-conftest.sh:28]: you are not calling this function neither usage

Comment (spiffxp) [reply hack/verify-conftest.sh:28]: Great catch thank you. I'll also try and fold in a commit that adds a failing policy and then fixes next commit

File: hack/verify-conftest.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pushd "${REPO_ROOT}" >/dev/null
+    local k8s_yaml_paths=(
+        apps/
+        infra/gcp/clusters/projects/*/*/resources/*.yaml
+    )
+    conftest test --policy policies/ "${k8s_yaml_paths[@]}"
+
+}
+
+main
```
Comment (cpanato) [inline hack/verify-conftest.sh:45]: nit: new line

**Comment by spiffxp** [comment]
I could not get as far as "writing a failing policy and then fix it".  What we have now is not quite as copy-paste friendly as a thought, and I'd rather not have this hang out until I more properly learn Rego.

I did at least add some warnings for v1.22 removals we'll want to address

**Pull request merged**